### PR TITLE
fix: `clipboard.readImage()` should be synchronous

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/39055
Refs CL:4216473

This PR makes `clipboard.readImage()` a synchronous function forcibly. This is now required as a result of CL:4216473, which moved png file encoding to a `gfx::Image` to a background thread. We didn't catch this in testing because [`EncodeGfxImageToPng`](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:ui/base/clipboard/clipboard_mac.mm;drc=fdfd85f836e0e59c79ed9bf6d527a2b8f7fdeb6e;bpv=1;bpt=1;l=494?q=GetPngFromPasteboard&sq=package:chromium&type=cs) is invoked in the case where a user for example copies an image in Finder using the context menu. Calling `clipboard.writeImage()` and then `clipboard.readImage()` as we do in tests would take the still-synchronous [`GetPngFromPasteboard`](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:ui/base/clipboard/clipboard_mac.mm;l=82;drc=fdfd85f836e0e59c79ed9bf6d527a2b8f7fdeb6e;bpv=1;bpt=1?q=GetPngFromPasteboard&sq=package:chromium&type=cs) pathway.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed a crash in 
`clipboard.readImage()` when reading images copied via context menu on macOS.

### CODE REVIEW

**Review by jkleinsc — COMMENTED** [review]
API LGTM

File: docs/api/native-image.md [hunk 1]
```diff
@@ -21,10 +21,13 @@ console.log(appIcon, win)
 Or read the image from the clipboard, which returns a `NativeImage`:
 
 ```javascript
-const { clipboard, Tray } = require('electron')
-const image = clipboard.readImage()
-const appIcon = new Tray(image)
-console.log(appIcon)
+const { app, clipboard, Tray } = require('electron')
+
+app.on('ready', async () => {
```
Comment (dsanders11) [inline docs/api/native-image.md:26]:
```suggestion
app.whenReady().then(async () => {
```

The docs prefer `whenReady()` over `on('ready', ...` these days.

**Review by erickzhao — APPROVED** [review]
API LGTM

**Review by zcbenz — COMMENTED** [review]
You can keep `readImage` a sync function by running a nested loop:

```c++
gfx::Image Clipboard::ReadImage() {
  base::RunLoop run_loop;
  base::RepeatingClosure callback = run_loop.QuitClosure();
  clipboard->ReadPng(
      base::BindOnce([callback]() {
        callback();
      }));
  run_loop.Run();
}
```

This may look horrible but it is actually used frequently in Chromium. For example, reading text from clipboard is async on Linux but encapsulated to be a sync method with this trick.

Even if it is decided to remove the sync version, the breaking change should be break things very loudly, silently changing the return value's type usually means very-difficult-to-debug bugs.

```js
 // Removed in Electron 26
 const image = clipboard.readImage()
 throw new Error('clipboard.readImage API has been removed')

 // Replace with
 clipboard.readImageAsync().then(image => {
   console.log(image)
 })

```

**Comment by codebytere** [comment]
@zcbenz that's a good point - we're ultimately moving in this direction: https://github.com/electron/governance/pull/454 which is async anyway but i agree i'd prefer not to break if reasonably avoidable.

**Comment by itsananderson** [comment]
> Release Notes
> Notes: A crash in clipboard.readImage() when reading images copied via context menu on macOS.

Should the release notes be worded more like "Fixed a crash..."?

**Comment by codebytere** [comment]
@itsananderson yes, my bad on that typo haha

File: docs/api/native-image.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ Or read the image from the clipboard, which returns a `NativeImage`:
 
 ```javascript
 const { clipboard, Tray } = require('electron')
+
```
Comment (jkleinsc) [inline docs/api/native-image.md:25]:
```suggestion
```

**Review by jkleinsc — APPROVED** [review]

**Comment by johnlindquist** [comment]
Thanks @codebytere !

**Pull request merged**